src/viam/sdk/components/movement_sensor/client.cpp

```diff
@@ -0,0 +1,121 @@
+#include <viam/sdk/components/movement_sensor/client.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/component/movementsensor/v1/movementsensor.grpc.pb.h>
+
+#include <viam/sdk/common/linear_algebra.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/components/movement_sensor/movement_sensor.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/robot/client.hpp>
+
+namespace viam {
+namespace sdk {
+
+MovementSensorClient::MovementSensorClient(std::string name, std::shared_ptr<grpc::Channel> channel)
+    : MovementSensor(std::move(name)),
+      stub_(viam::component::movementsensor::v1::MovementSensorService::NewStub(channel)),
+      channel_(std::move(channel)){};
+
+/// helper to throw on !ok grpc status
+void throw_status(const ::grpc::Status& status) {
```

Comment:
Just a note, but one problem with functions like this is that it means that the throw location is always this function, and not the place where the error happened. That's why these are often implemented via macros.

Reply:
made macro in 3d47fa0